apps/prairielearn/src/lib/db-types.ts

```diff
@@ -585,6 +586,16 @@ export const GradingJobSchema = z.object({
 });
 export type GradingJob = z.infer<typeof GradingJobSchema>;
 
+export const ClientFingerprintSchema = z.object({
+  id: IdSchema,
+  user_id: IdSchema,
+  user_session_id: IdSchema,
+  ip_address: z.string().nullable(),
```

Comment:
It looks like this is marked as `NOT NULL` in the DB schema?

Reply:
Good catch. I have now removed `.nullable()` from `ip_address` in the zod schema in commit [79aff64](https://github.com/PrairieLearn/PrairieLearn/pull/8798/commits/79aff648f5589ef70706839320ae5075eb431f6d).